Jenkinsfile

```
@@ -783,9 +789,14 @@ def makeTarget(String context, String target, boolean clean = true) {
   }
 }
 
-def mageTarget(String context, String directory, String target) {
+def mageTarget(Map args = [:]) {
+  def context = args.context
+  def directory = args.directory
+  def target = args.target
+  def withModule = args.get('withModule', false)
+  def modulePattern = args.get('modulePattern', '')
```

Could the module pattern be formed from the `directory`? So we don't need to pass an special module pattern for x-pack beats.

Would something like this work?
```suggestion
  def customModulePattern = args.get('modulePattern', 'module\\/([^\\/]+)\\/.*')
  def modulePattern = "^" + directory + "/" + customModulePattern
```

---

That's indeed a good catch, my only concern might be the path separator should be escaped in the directory, Let me give a go if I can do the transformation easily

---

I think I've just found the pattern:
- oss modules don't use the `mageTarget` but `makeTarget`
- x-pack modules use the `mageTarget`.

If that's the pattern then `modulePattern` is not anymore required as a parameter but set in:
- `makeTarget`
- `mageTarget`
- `mageTargetWin`

---

See https://github.com/elastic/beats/pull/18777/commits/c64efccf8b8f5289fa717642270c06d9446db454

---

> I think I've just found the pattern:
>     * oss modules don't use the `mageTarget` but `makeTarget`
>     * x-pack modules use the `mageTarget`.

This is casual at the moment. The idea is to migrate everything to mage. Actually there are oss beats using mage, only not with modules. Also all Windows stages use mage.

If we cannot use the directory to build the regular expression, maybe we can decide what regular expression to use depending on the directory. If it starts with `x-pack` then we use the xpack pattern. Wdyt about this option?

---

Ok, so if we use mage then the directory argument is required, while using make it's not required as we use the target with `-C folder. 

For instance:

```
mageTarget(context: "Elastic Agent x-pack Linux", directory: "x-pack/elastic-agent", target: "build test")

makeTarget(context: "Heartbeat oss Linux", target: "-C heartbeat testsuite")
```

So I'll use the parsing as suggested in the mage to support the upcoming mage only builds. And for the make I'll do the same for the target parameter if it contains `x-pack`

---

See https://github.com/elastic/beats/pull/18777/commits/a449455d0fdd15e429c315c19fb3372ff2e4b3aa

I've decided to use `contains` to refactor the logic to one function, if there are no more `x-pack` folders then we are safe otherwise, I might need to change the above to be more specific for the `makeTarget`

---

:+1: